sonarcloud/resource_quality_gate_selection.go

```diff
@@ -0,0 +1,270 @@
+package sonarcloud
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/reinoudk/go-sonarcloud/sonarcloud/qualitygates"
+)
+
+type resourceQualityGateSelectionType struct{}
+
+func (r resourceQualityGateSelectionType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "This resource selects a quality gate for a project",
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:        types.StringType,
+				Description: "The implicit ID of the resource",
+				Computed:    true,
+			},
+			"gate_id": {
+				Type:        types.StringType,
+				Description: "The ID of the quality gate that is selecting the project(s).",
+				Required:    true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.RequiresReplace(),
+				},
+			},
+			"project_key": {
```

Comment:
I'd use the plural `project_keys` here, as it is a set of project keys, not a single project key.